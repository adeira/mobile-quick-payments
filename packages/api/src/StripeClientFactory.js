// @flow

import Stripe from 'stripe';

/**
 * TODOs:
 *
 * - https://stripe.com/docs/api#errors
 * - https://stripe.com/docs/api#idempotent_requests
 */
export default class StripeClientFactory {
  apiToken: string;
  apiVersion: string;

  constructor(
    apiToken: string = String(process.env.STRIPE_SECRET_API_TOKEN),
    apiVersion: string = '2018-02-28',
  ) {
    this.apiToken = apiToken;
    this.apiVersion = apiVersion;
  }

  createClient() {
    return new Stripe(this.apiToken, this.apiVersion);
  }
}
