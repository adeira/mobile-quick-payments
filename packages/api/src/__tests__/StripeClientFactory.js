// @flow

import StripeClientFactory from '../StripeClientFactory';

it('creates Stripe client', () => {
  expect(
    new StripeClientFactory('SECRET API TOKEN').createClient()
  ).toMatchSnapshot();
});
