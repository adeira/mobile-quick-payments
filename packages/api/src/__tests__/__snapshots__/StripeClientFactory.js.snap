// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates Stripe client 1`] = `
Object {
  "_api": Object {
    "agent": null,
    "auth": "Bearer SECRET API TOKEN",
    "basePath": "/v1/",
    "dev": false,
    "host": "api.stripe.com",
    "port": "443",
    "timeout": 120000,
    "version": "2018-02-28",
  },
  "account": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "path": [Function],
    "resourcePath": "",
  },
  "accounts": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "path": [Function],
    "resourcePath": "",
  },
  "applePayDomains": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "del": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "apple_pay/domains",
    "retrieve": [Function],
  },
  "applicationFeeRefunds": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "application_fees/{feeId}/refunds",
    "retrieve": [Function],
    "update": [Function],
  },
  "applicationFees": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "application_fees",
    "retrieve": [Function],
  },
  "balance": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "path": [Function],
    "resourcePath": "balance",
  },
  "bitcoinReceivers": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "getMetadata": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "bitcoin/receivers",
    "retrieve": [Function],
    "setMetadata": [Function],
    "update": [Function],
  },
  "chargeRefunds": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "charges/{chargeId}/refunds",
    "retrieve": [Function],
    "update": [Function],
  },
  "charges": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "getMetadata": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "charges",
    "retrieve": [Function],
    "setMetadata": [Function],
    "update": [Function],
  },
  "countrySpecs": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "country_specs",
    "retrieve": [Function],
  },
  "coupons": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "del": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "coupons",
    "retrieve": [Function],
    "update": [Function],
  },
  "customerCards": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "del": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "customers/{customerId}/cards",
    "retrieve": [Function],
    "update": [Function],
  },
  "customerSubscriptions": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "del": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "customers/{customerId}/subscriptions",
    "retrieve": [Function],
    "update": [Function],
  },
  "customers": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "del": [Function],
    "getMetadata": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "customers",
    "retrieve": [Function],
    "setMetadata": [Function],
    "update": [Function],
  },
  "disputes": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "getMetadata": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "disputes",
    "retrieve": [Function],
    "setMetadata": [Function],
    "update": [Function],
  },
  "ephemeralKeys": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "del": [Function],
    "path": [Function],
    "resourcePath": "ephemeral_keys",
  },
  "events": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "events",
    "retrieve": [Function],
  },
  "exchangeRates": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "exchange_rates",
    "retrieve": [Function],
  },
  "fileUploads": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "files",
    "retrieve": [Function],
  },
  "invoiceItems": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "del": [Function],
    "getMetadata": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "invoiceitems",
    "retrieve": [Function],
    "setMetadata": [Function],
    "update": [Function],
  },
  "invoices": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "invoices",
    "retrieve": [Function],
    "update": [Function],
  },
  "loginLinks": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "path": [Function],
    "resourcePath": "accounts/{accountId}/login_links",
  },
  "off": [Function],
  "on": [Function],
  "orderReturns": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "order_returns",
    "retrieve": [Function],
  },
  "orders": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "orders",
    "retrieve": [Function],
    "update": [Function],
  },
  "payouts": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "getMetadata": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "payouts",
    "retrieve": [Function],
    "setMetadata": [Function],
    "update": [Function],
  },
  "plans": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "del": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "plans",
    "retrieve": [Function],
    "update": [Function],
  },
  "products": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "del": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "products",
    "retrieve": [Function],
    "update": [Function],
  },
  "recipientCards": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "del": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "recipients/{recipientId}/cards",
    "retrieve": [Function],
    "update": [Function],
  },
  "recipients": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "del": [Function],
    "getMetadata": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "recipients",
    "retrieve": [Function],
    "setMetadata": [Function],
    "update": [Function],
  },
  "refunds": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "refunds",
    "retrieve": [Function],
    "update": [Function],
  },
  "skus": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "del": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "skus",
    "retrieve": [Function],
    "update": [Function],
  },
  "sources": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "getMetadata": [Function],
    "path": [Function],
    "resourcePath": "sources",
    "retrieve": [Function],
    "setMetadata": [Function],
    "update": [Function],
  },
  "subscriptionItems": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "del": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "subscription_items",
    "retrieve": [Function],
    "update": [Function],
  },
  "subscriptions": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "del": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "subscriptions",
    "retrieve": [Function],
  },
  "threeDSecure": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "path": [Function],
    "resourcePath": "3d_secure",
    "retrieve": [Function],
  },
  "tokens": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "path": [Function],
    "resourcePath": "tokens",
    "retrieve": [Function],
  },
  "topups": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "getMetadata": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "topups",
    "retrieve": [Function],
    "setMetadata": [Function],
    "update": [Function],
  },
  "transferReversals": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "transfers/{transferId}/reversals",
    "retrieve": [Function],
    "update": [Function],
  },
  "transfers": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "create": [Function],
    "getMetadata": [Function],
    "list": [Function],
    "path": [Function],
    "resourcePath": "transfers",
    "retrieve": [Function],
    "setMetadata": [Function],
    "update": [Function],
  },
  "usageRecords": Object {
    "_stripe": [Circular],
    "_urlData": Object {},
    "basePath": [Function],
    "path": [Function],
    "resourcePath": "subscription_items",
  },
  "webhooks": Object {
    "DEFAULT_TOLERANCE": 300,
    "constructEvent": [Function],
    "signature": Object {
      "EXPECTED_SCHEME": "v1",
      "_computeSignature": [Function],
      "verifyHeader": [Function],
    },
  },
}
`;
